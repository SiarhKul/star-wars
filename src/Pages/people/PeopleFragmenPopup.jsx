import React from "react";
import { useSelector } from "react-redux";
import { PopupListItem } from "../../components/popup/PopupListItem";

export const PeopleFragmenPopup = () => {
	const {
		name,
		hair_color,
		skin_color,
		eye_color,
		gender,
		height,
		mass,
		homeworld,
		films,
		vehicles,
		starships,
	} = useSelector(state => state.app.clickedCard);

	return (
		<>
			<div className="name-wrapper">
				<span className="popup__name">{name}</span>
			</div>
			<div className="description-wrapper">
				<div className="description-person">
					<div className="description-appearance">
						<p className="description__heading">Appearance</p>
						<ul className="description-appearance__item">
							<li>Hair color:{hair_color}</li>
							<li>Skin color:{skin_color}</li>
							<li>Eye color: {eye_color}</li>
							<li>Gender:{gender}</li>
						</ul>
					</div>

					<div className="description-stats">
						<p className="description__heading">Stats</p>
						<ul className="description-stats__items">
							<li>Height:{height} cm</li>
							<li>Mass:{mass} kg</li>
						</ul>
					</div>
				</div>

				<div className="description-resources">
					<div className="resources-homeworld">
						<div className="resources-heading">Home word:</div>
						<ul className="resources-list">
							<PopupListItem urls={[homeworld]} name="name" />
						</ul>
					</div>

					<div className="resources-films">
						<div className="resources-heading">Films:</div>
						<ul className="resources-list">
							<PopupListItem urls={films} name="title" />
						</ul>
					</div>

					<div className="resources-vehicles">
						<div className="resources-heading">Vehicles:</div>
						<ul className="resources-list">
							<PopupListItem urls={vehicles} name="name" />
						</ul>
					</div>

					<div className="resources-starships">
						<span className="resources-heading">Starships:</span>
						<ul className="resources-list">
							<PopupListItem urls={starships} name="name" />
						</ul>
					</div>
				</div>
			</div>
		</>
	);
};
