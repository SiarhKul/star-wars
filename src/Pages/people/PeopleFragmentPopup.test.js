import { shallow } from "enzyme";
import React from "react";
import { PeopleFragmenPopup } from "./PeopleFragmenPopup";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => {
			return {
				location: {
					state: {
						name: "Luke Skywalker",
						height: "172",
						mass: "77",
						hair_color: "blond",
						skin_color: "fair",
						eye_color: "blue",
						birth_year: "19BBY",
						gender: "male",
						homeworld: "https://swapi.dev/api/planets/1/",
						films: ["https://swapi.dev/api/films/6/"],
						species: [],
						vehicles: ["https://swapi.dev/api/vehicles/30/"],
						starships: ["https://swapi.dev/api/starships/22/"],
					},
				},
			};
		}),
	};
});

describe("Test <PeopleFragmentPopup/>", () => {
	const peopleFragmentPopup = () => shallow(<PeopleFragmenPopup />);
	const component = peopleFragmentPopup();

	it("render snapshot <PeopleFragmentPopup/>", () => {
		expect(component).toMatchSnapshot();
	});

	it("should <PeopleFragmentPopup/> have heading", () => {
		const requiredFields = [
			"Appearance",
			"Hair color:",
			"Skin color:",
			"Eye color:",
			"Gender:",
			"Stats",
			"Height:",
			"Mass:",
			"Films:",
			"Vehicles:",
			"Starships:",
		];

		expect(component.containsAllMatchingElements(requiredFields)).toBeTruthy();
	});
});
