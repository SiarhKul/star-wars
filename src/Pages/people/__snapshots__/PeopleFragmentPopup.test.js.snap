// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PeopleFragmentPopup/> render snapshot <PeopleFragmentPopup/> 1`] = `
<Fragment>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    >
      Luke Skywalker
    </span>
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Hair color:
            blond
          </li>
          <li>
            Skin color:
            fair
          </li>
          <li>
            Eye color:
            blue
          </li>
          <li>
            Gender:
            male
          </li>
        </ul>
      </div>
      <div
        className="description-stats"
      >
        <p
          className="description__heading"
        >
          Stats
        </p>
        <ul
          className="description-stats__items"
        >
          <li>
            Height:
            172
             cm
          </li>
          <li>
            Mass:
            77
             kg
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources resources_no-list-type"
      >
        <div
          className="resources-heading"
        >
          Home word:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="planets"
            urls={
              Array [
                "https://swapi.dev/api/planets/1/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Films:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="title"
            path="films"
            urls={
              Array [
                "https://swapi.dev/api/films/6/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Vehicles:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="vehicles"
            urls={
              Array [
                "https://swapi.dev/api/vehicles/30/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <span
          className="resources-heading"
        >
          Starships:
        </span>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="starships"
            urls={
              Array [
                "https://swapi.dev/api/starships/22/",
              ]
            }
          />
        </ul>
      </div>
    </div>
  </div>
</Fragment>
`;
